Source/Platform/PlatformComponents.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "IHttpProvider.h"
+#include "IWebSocketProvider.h"
+
+NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
+
+// Container for all Platform specific components needed by libHttpClient.
+// Implementations for each of these interfaces is needed for each supported platform.
+// Additionally, each platform must implement its own version of PlatformInitialize.
+struct PlatformComponents
+{
+    HC_UNIQUE_PTR<IHttpProvider> HttpProvider;
+    HC_UNIQUE_PTR<IWebSocketProvider> WebSocketProvider;
```

Comment:
Going to leave these off for since PlayFab doesn't care about HC_NO_WEBSOCKETS.  Will go back and add them before merging back to main if you'd like

Reply:
It would be useful to preserve HC_NO_WEBSOCKETS for the integration into GRTS where we want to minimize dependencies, I'm good with it being readded later, as long as we don't make it impossible to do